packages/expect/src/utils.ts

```diff
@@ -251,6 +251,14 @@ export const iterableEquality = (
     return false;
   }
 
+  if (!emptyObject(a) && !emptyObject(b)) {
```

Comment:
Why do we need the `emptyObject` checks? If `Object.entries` returns empty arrays that'd be fine, wouldn't it?

Reply:
i was thinking of avoiding `equals` call with an empty array, but yes `Object.entries` is enough

Reply:
I think I'd favor readability over performance here, `iterableEquality` is already terribly complex 😅